src/app/core/data/request.service.ts

```diff
@@ -147,21 +148,23 @@ export class RequestService {
    * @param {RestRequest} request The request to send out
    */
   configure<T extends CacheableObject>(request: RestRequest): void {
-    const isGetRequest = request.method === RestRequestMethod.GET;
-    if (!isGetRequest || request.forceBypassCache || !this.isCachedOrPending(request)) {
-      this.dispatchRequest(request);
-      if (isGetRequest) {
-        this.trackRequestsOnTheirWayToTheStore(request);
-      }
-    } else {
-      this.getByHref(request.href).pipe(
-        filter((entry) => hasValue(entry)),
-        take(1)
-      ).subscribe((entry) => {
-          return this.store.dispatch(new AddToIndexAction(IndexName.UUID_MAPPING, request.uuid, entry.request.uuid))
+    this.zone.runOutsideAngular(() => {
```

Comment:
Could we add a comment here as to why this block of code needs to be run outside of Angular's zone? It'd be very helpful to other developers looking at this code section later on.

Reply:
It doesn't need to be, but it can be, and It could be a performance benefit. I'll add some more comments.

Reply:
@artlowel : Thanks, yes I would appreciate comments here so that we all understand why this was added -- as it's unusual for the UI to run anything outside of Angular's zone.

Reply:
added

Reply:
@artlowel : I'm not seeing the added commit.  Did you push it up to GitHub yet?